### PR TITLE
Make package v2.0

### PR DESCRIPTION
**Pull request by apulsipher**
This branch (`make-package-v2.0`) enables package installation for `multitask_unit_and_days` using `pip install` with the Github URL.

The main issue was with file paths:
- I moved `multitask_unit_and_days` up to the root level (deleting `src/`)
- I updated the internal import statements in files such as `run_experiments.py` to explicitly import from `multitask_unit_and_days` using syntax `import multitask_unit_and_days.<module_name>` (e.g., `import multitask_unit_and_days.DataManager as dm`). 

Other notes:
- I also commented out the contents of `__init__.py`, but that was before fixing the file paths, so I don't think that helped any. In any case, it clearly isn't necessary to install the package and get it working, though it's possible the file could have some uses that I'm not familiar with.

Remaining issues:
- There are some hacky steps to get it running in Podman container that can be streamlined (I lack familiarity with Podman and containerization)
- The package requires `distutils`
- Import statements could use `from multitask_unit_and_days import <module_name>` instead of `import multitask_unit_and_days.<module_name>`
- Decide if we need `__init__.py` or if it can be deleted. It clearly isn't necessary to install the package, but maybe it can serve some other useful function.
- Documentation needs to be clarified and updated, especially for detailing the process of running the tool on the example/dummy dataset provided
- Poetry could be leveraged more to build the package and store it on PyPI, which would give more installation options for users
- While the classes and functions within the package work, the files `run_experiments.py` and `main.py` are built to run on datasets in the `data` folder. If this package is to be generally available, we need to know if those datasets need to be included with the package or if these functions need to be updated to run generically for any dataset the user provides. That will require updating the `config.py` file as well, where most of these file paths are stored. **This is the biggest component of the work needed to make this package ready for general distribution.**

### CODE REVIEW

**Comment by apulsipher** [comment]
@gvegayon I've added you to this pull request so you can check it out. Let me know your thoughts on the "Remaining issues" I outlined above. If you agree, I can go ahead and start on those (if we don't have anything better for me to do at the moment), otherwise I will generate them as issue tickets so we don't forget they need to be worked on.

**Comment by apulsipher** [comment]
Also, we discussed in our standup meeting emailing Ananth (the project's key contact from the tools spreadsheet in Box) about getting an example run. I see now that they have a dummy dataset for just that purpose. I'll add to "Remaining issues" to document the process of running that example, though it will need to be addressed after we decide what to do with the `run_experiments.py` and `main.py` files.

**Comment by gvegayon** [comment]
Thank you, @apulsipher! On the remainder issues:

> There are some hacky steps to get it running in Podman container that can be streamlined (I lack familiarity with Podman and containerization)

We don't need that here. But that's a good topic to include in the EpiForeSITE/software documentation repo.

> The package requires distutils

Do you need to do something other than add it to `pyproject.toml`?

> Import statements could use from multitask_unit_and_days import <module_name> instead of import multitask_unit_and_days.<module_name>

I'm fine with either. I think it is up to the user, no?

> Decide if we need __init__.py or if it can be deleted. It clearly isn't necessary to install the package, but maybe it can serve some other useful function.

I think it is OK not to have them? But I may be wrong.

> Documentation needs to be clarified and updated, especially for detailing the process of running the tool on the example/dummy dataset provided

For sure, but that's an issue to be posted as the next steps, I believe.

> Poetry could be leveraged more to build the package and store it on PyPI, which would give more installation options for users

Probably not for this project, but it will be good to have it documented with a sort of template, as you suggested.

> While the classes and functions within the package work, the files run_experiments.py and main.py are built to run on datasets in the data folder. If this package is to be generally available, we need to know if those datasets need to be included with the package or if these functions need to be updated to run generically for any dataset the user provides. That will require updating the config.py file as well, where most of these file paths are stored. This is the biggest component of the work needed to make this package ready for general distribution.

I agree; we should wait until we get the original dev to take it further.

File: README.md [hunk 1]
```diff
@@ -26,4 +26,17 @@ The starting point of the code is main.py. It takes the user input and then goes
 
 Since there are two different types of tasks, our evaluation metrics are set accordingly. Prediction of next unit label is a classification task, our evaluation metrics for this task are accuracy and top-k accuracy (k=2). Prediction of remaining length of stay is a regression task, our evaluation metrics in this case is mean absolute error, mean squared error and root mean squared error. The MAE and Top-k accuracy values are plotted and saved in the image folder to show the model performance on test data. The history of the training time is saved in the hist_log folder.
 
+### Running in Podman
+- Run a container using the `docker.io/library/python:3.12` image
+- By default, the image doesn't have access to a few supporting tools needed to install the package. One of these, h5py must be installed before installing the package. Run these commands in the container Terminal:
+  ```
+  apt update
+  apt-get install libhdf5-dev
+  ```
```
Comment (gvegayon) [inline README.md:35]: For this, we could use the `.devcontainer` feature ([here](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers)). This then allows firing up the repo from codespaces and/or VS code.

Comment (gvegayon) [reply README.md:35]: Certainly for the "template" project, but not necessarily for this project.

File: README.md [hunk 2]
```diff
@@ -26,4 +26,17 @@ The starting point of the code is main.py. It takes the user input and then goes
...
+- Then install the package using `pip install multitask_unit_and_days@git+https://github.com/EpiForeSITE/multitask_unit_and_days`
+  - To install package from a specific branch use `pip install multitask_unit_and_days@git+https://github.com/EpiForeSITE/multitask_unit_and_days@<branch_name>`
+    - For example, to install the package version from branch `make-package-v2.0` use: `pip install multitask_unit_and_days@git+https://github.com/EpiForeSITE/multitask_unit_and_days@make-package-v2.0`
+- The next tool needed is `distutils`, which is not imported automatically. Import this in the container's Python Tty environment using `import setuptools.dist`
```
Comment (gvegayon) [inline README.md:39]: Why is that the case? Shouldn't you add this to the pyproject.toml file?

**Review by gvegayon — APPROVED** [review]
This is looking great, @apulsipher! I haven't tried it on my own, but maybe we could add a GitHub action that tries installing the package via pip using an empty runner. You could even use the `docker.io/library/python:3.12` image. Can you create an issue so we can address it there?

**Comment by apulsipher** [comment]
> This is looking great, @apulsipher! I haven't tried it on my own, but maybe we could add a GitHub action that tries installing the package via pip using an empty runner. You could even use the `docker.io/library/python:3.12` image. Can you create an issue so we can address it there?

Yes, I'll add that as a separate issue.

File: README.md (hunk 2 above)
Comment (apulsipher) [reply README.md:39]: That's a good point, thanks. I'll give that a try.

**Comment by apulsipher** [comment]
I'm going to merge this branch and then address the above-mentioned items in separate issues/branches

**Pull request merged**